### ISSUE TITLE
JwtBearerEvents might be null, despite being marked as not null

### ISSUE THREAD

**Issue by glatzert** [dotnet/aspnetcore#43313]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The following code will throw a NullReferenceException:
```
services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
                .AddJwtBearer(o =>
                {
                    o.Events.OnMessageReceived += (ctx) => Task.CompletedTask;
                });
```

It will do so despite o.Events being flagged as not nullable.

### Expected Behavior

o.Events is not `null`

### .NET Version

6.0.400

### Anything else?

https://github.com/dotnet/aspnetcore/blob/cccab50fba1c7335fdb48a735246374b5ddd7e69/src/Security/Authentication/JwtBearer/src/JwtBearerOptions.cs#L57-L66

This should probably have `JwtBearerEvents?`

**Comment by Tratcher** [comment]
You're correct, the base Events field is nullable and so should this one be. Note this applies to all auth handlers.
https://github.com/dotnet/aspnetcore/blob/cccab50fba1c7335fdb48a735246374b5ddd7e69/src/Security/Authentication/Core/src/AuthenticationSchemeOptions.cs#L33

Events will be initialized to a default value later if one is not provided.
https://github.com/dotnet/aspnetcore/blob/c85baf8db0c72ae8e68643029d514b2e737c9fae/src/Security/Authentication/Core/src/AuthenticationHandler.cs#L154

**Comment by akonyer** [comment]
I'm running into this as well. For now just have to null check it even though the compiler tells me you shouldn't have to
